conans/model/build_info.py

```diff
@@ -1,224 +1,620 @@
-        self.libdirs.append(DEFAULT_LIB)
-        self.bindirs.append(DEFAULT_BIN)
-        self.resdirs.append(DEFAULT_RES)
-        self.builddirs.append("")
-        # public_deps is needed to accumulate list of deps for cmake targets
-        self.public_deps = []
+    List of build information about each of the nodes in the graph.
+    It contains the flags, directories and options if its dependencies.
+    Provides properties to access aggregated information of directories and flags.
+    The access to the information of each node can also be accessed.
+    Should use these flags to pass them to the underlaying build system (Cmake, make),
+    so deps info is managed.
+    """
+
+    def __init__(self):
+        self._dependencies = OrderedDict()
         self.configs = {}
 
     def __getattr__(self, config):
+        # If the configuration does not exist, return an empty list
+        # FIXME: This could create unintended empty configurations for those generators/libs
+        # that access unexisting configs with self.deps_cpp_info.whatever.includedirs
+        return self.configs.setdefault(config, DepsCppInfo())
 
-        def _get_cpp_info():
-            result = _CppInfo()
-            result.rootpath = self.rootpath
-            result.sysroot = self.sysroot
-            result.includedirs.append(DEFAULT_INCLUDE)
-            result.libdirs.append(DEFAULT_LIB)
-            result.bindirs.append(DEFAULT_BIN)
-            result.resdirs.append(DEFAULT_RES)
-            result.builddirs.append("")
-            return result
+    @property
+    def dependencies(self):
+        return self._dependencies.items()
 
-        return self.configs.setdefault(config, _get_cpp_info())
+    @property
+    def deps(self):
+        return self._dependencies.keys()
 
+    def __getitem__(self, item):
+        return self._dependencies[item]
 
-class _BaseDepsCppInfo(_CppInfo):
-    def __init__(self):
-        super(_BaseDepsCppInfo, self).__init__()
+    def update(self, cpp_info, pkg_name):
+        assert isinstance(cpp_info, CppInfo)
+        self.update_dep_cpp_info(DepCppInfo(cpp_info), pkg_name)
 
-    def update(self, dep_cpp_info):
+    def update_dep_cpp_info(self, dep_cpp_info, pkg_name):
+        assert isinstance(dep_cpp_info, DepCppInfo)
+        self._dependencies[pkg_name] = dep_cpp_info
+        if dep_cpp_info.configs:
+            for config, sub_dep_cpp_info in dep_cpp_info.configs.items():
+                self.configs.setdefault(config, DepsCppInfo()).update_dep_cpp_info(sub_dep_cpp_info,
+                                                                                   pkg_name)
 
-        def merge_lists(seq1, seq2):
-            return [s for s in seq1 if s not in seq2] + seq2
+    @property
+    def version(self):
+        return None  # Backwards compatibility: Do not brake scons generator
 
-        self.includedirs = merge_lists(self.includedirs, dep_cpp_info.include_paths)
```

Comment:
There were reasons this was computed here, they were being cached. This access can be done a few times from the code, and merging those lists is not that cheap.

Reply:
Made the changes so values are initialized and cached from the first moment. Thanks!